app/routes/api.js

```diff
@@ -27,15 +57,24 @@ function query(query, values, res, sendStatus = false) {
         });
 }
 
-// TODO: put auth functions and other stuff into a seperate file
-const tokenStorage = {};
 const authentication = (req, res, next) => {
     const { posAuth } = req.cookies;
 
-    if (!tokenStorage.hasOwnProperty(posAuth) || !tokenStorage[posAuth])
-        return res.sendStatus(401);
-
-    next();
+    pool.query("SELECT * FROM tokens WHERE token = $1", [posAuth])
+        .then((result) => {
+            if (result.rows.length === 0) {
+                // redirect to login page
+                // https://expressjs.com/en/4x/api.html#res.redirect
+                return res.redirect("/pos/login");
+            }
+
+            // not putting next() in here caused me an hour of pain
+            next();
```

Comment:
what does next do

Reply:
pass it off to the request handler or middleware